### PR TITLE
feat: add dashboard charts and maps

### PR DESCRIPTION
**Pull request by paigen11**
# Problem Context

Currently our airnote.live dashboards link to Grafana for “full data”. That full data includes charts for air quality, temperature, humidity, barometric pressure, and voltage. Here’s an example: [Grafana](https://grafana.safecast.cc/d/7wsttvxGk/safecast-airnote?orgId=1&var-device_urn=note:dev:864475044221348).

Grafana allows users to configure their date range, and they allow users to show years' worth of data if they’d like. We probably can’t totally replicate that at the moment because of Notehub API limitations, but we should be able to provide seven days worth of data.

## Changes

* Added charts to display temperature, voltage and charging periods, humidity, AQI levels, and PM readings.
* Added a map displaying current Airnote location, current AQI level, and PM levels in a popup when the marker is clicked.
* Updated the AQI speedometer to match US AQI colors and level classifications.
* Grouped together current and historical readings in the dashboard to make it easier for users to know what type of data they're viewing.
* Removed links to Safecast Grafana dashboards and charts.
* Added Cypress e2es that can be run locally to ensure all pieces of Airnote are working / rendering as expected.

## Screenshot (if applicable)

https://user-images.githubusercontent.com/20400845/183437514-791bfe64-39bf-4226-bcea-a361e5ea5b95.mov

## Testing

Unit / integration / e2e tests?
Steps to test manually?
Any other sorts of testing notes to include?

## Any other related PRs

https://github.com/blues/airnote.live/pull/43 - TJ's PR must merge first in order for my PR here to work properly

## Ticket(s)

https://trello.com/c/GfwLHF1H/58-replicate-grafana-charts-on-airnotelive
https://trello.com/c/E4huY0UC/87-add-airnote-ci-cd-pipeline-and-e2es

### CODE REVIEW

**Comment by paigen11** [comment]
[Replicate Grafana charts on airnote.live](https://trello.com/c/GfwLHF1H/58-replicate-grafana-charts-on-airnotelive)

**Comment by paigen11** [comment]
[Add Airnote CI/CD pipeline and e2es](https://trello.com/c/E4huY0UC/87-add-airnote-ci-cd-pipeline-and-e2es)

File: src/routes/Dashboard/History.svelte [hunk 1]
```diff
@@ -79,29 +101,29 @@
   }
   .history {
     display: grid;
-    grid-template-columns: 14.28% 14.28% 14.28% 14.28% 14.28% 14.28% 14.28%;
+    grid-template-columns: repeat(7, 1fr);
```
Comment (tjvantoll) [inline src/routes/Dashboard/History.svelte:104]: today I learned

File: src/routes/Dashboard/Dashboard.svelte [hunk 2]
```diff
@@ -77,15 +82,15 @@
           // This is a hack, but the speedometer plugin doesn’t give any
           // way to customize these labels.
           setTimeout(() => {
```
Comment (tjvantoll) [inline src/routes/Dashboard/Dashboard.svelte:84]: I think you can remove this entire setTimeout hack to set the label. AQI values can’t be larger than 500, so there’s no need to change "500" --> "500+".

File: src/routes/Dashboard/Dashboard.svelte [hunk 3]
```diff
@@ -432,12 +466,66 @@
...
+  }
+
+  .chart3 {
+    grid-area: chart3;
+    max-width: 475px;
+  }
+
+  .chart4 {
+    grid-area: chart4;
+    max-width: 475px;
```
Comment (tjvantoll) [inline src/routes/Dashboard/Dashboard.svelte:499]: Super trivial, but might be worth consolidating these `475px` properties into one rule.

File: src/services/air.js [hunk 4]
```diff
@@ -1,85 +1,148 @@
...
+  "#7e0023",
+];
+
+export const aqiRanges = [
+  "0 - 50",
+  "51-100",
+  "101-150",
+  "151-200",
+  "201-300",
+  "301 - 500+",
```
Comment (tjvantoll) [inline src/services/air.js:33]: Odd mix of spacing between numbers and dashes here.

Also I think the `+` can go here as well. Basing off https://www.epa.gov/outdoor-air-quality-data/air-data-basic-information.

File: src/services/air.js [hunk 5]
```diff
@@ -1,85 +1,148 @@
...
+];
 
 export function getDisplay(type, value) {
-  if (type == 'aqi') {
+  if (type == "aqi") {
     return getAQIDisplay(value);
-  } else if (type == 'pm2_5') {
-    return getPM2_5Display(value);
+    // } else if (type == "pm2_5") {
+    //   return getPM2_5Display(value);
```
Comment (tjvantoll) [inline src/services/air.js:51]: Can probably just remove the commented code.

File: src/services/air.js [hunk 6]
```diff
@@ -1,85 +1,148 @@
...
-    case (value > 120):
-      return VERY_UNHEALTHY
-    case (value > 80):
-      return UNHEALTHY_SENSITIVE;
-    case (value > 40):
-      return MODERATE;
-    default:
-      return GOOD;
-  }
+// Based on https://blissair.com/what-is-pm-2-5.htm
```
Comment (tjvantoll) [inline src/services/air.js:98]: I’d just kill all the commented stuff in this file.

**Review by tjvantoll — APPROVED** [review]
Looks awesome! I only noted a few trivial things.

One suggestion: it might be nice to have all the chart components in their own folder, as the `Dashboard` folder is starting to get a bit cluttered.

File: src/routes/Dashboard/Dashboard.svelte (hunk 2 above)
Comment (paigen11) [reply src/routes/Dashboard/Dashboard.svelte:84]: Good call! I didn't realize 500 was our ceiling

File: src/services/air.js (hunk 4 above)
Comment (paigen11) [reply src/services/air.js:33]: Good catch. I've codified

File: src/services/air.js (hunk 5 above)
Comment (paigen11) [reply src/services/air.js:51]: Yes, thanks!

File: src/services/air.js (hunk 6 above)
Comment (paigen11) [reply src/services/air.js:98]: Done

File: src/routes/Dashboard/Dashboard.svelte (hunk 3 above)
Comment (paigen11) [reply src/routes/Dashboard/Dashboard.svelte:499]: done

**Comment by paigen11** [comment]
Closing as we recreated this PR in a new version to test staging with render

**Pull request closed without merging**